### ISSUE TITLE
Add separate embeddings for individual networks to MASE and MASE tutorial

### ISSUE THREAD

**Issue by loftusa** [microsoft/graspologic#882]
Right now, MASE says that $A^{(i)} = VR^{(i})V^\T$, which means that the estimated latent position for each graph is $X^{(i)} = V S^{(i)}$ with $S^{(i)} = |R^{(i)}|^{1/2} = QD^{1/2}$, with Q and D obtained from the SVD of $R^{(i)}$. 

Currently, MASE doesn't return individual latent positions $X^{(i)}$ for each network with the `fit_transform` method, it returns $V$. Since it also has a `scores_` attribute, you can technically calculate each $X^{(i)}$, but I think it's a big ask to assume that the user of the library knows how to do that. 

I propose adding a `graph_latents_` attribute (or something similarly named), which is calculated lazily (e.g., calculations happen only when you need, via an `@property` decorator), and which retrieves the individual latent positions for each graph $X^{(i)} = V S^{(i)}$.

That way, it's also bit less overhead to compare directly with OMNI, since OMNI returns its version of the $X^{(i)}$ with its `fit_transform` by default.

Thoughts?

**Comment by bdpedigo** [comment]
im fine with something like this, think it would be useful to have. i have code somewhere. not sure what the API should look like though, are you saying fit_transform should return this instead? 

also, there is a nonidentifiability associated with MASE if you are doing per-vertex-per-graph representations which was not brought up in either the COSIE (since they didnt even look at this) or time series anomaly paper. TLDR: there's nothing saying that the SVDs of the R matrices for each graph will be aligned, so if you do X^i = V @ S^i, there Xs will inherit this nonidentifiability.

**Comment by loftusa** [comment]
> are you saying fit_transform should return this instead?

No, I think we should just add a lazy attribute to the class instantiation object that appears after you call `fit`

> TLDR: there's nothing saying that the SVDs of the R matrices for each graph will be aligned, so if you do X^i = V @ S^i, the Xs will inherit this nonidentifiability.

Wait really? That changes things, I thought they'd be in the same space like OMNI. Why would you do it that way instead of just individual ASE?

**Comment by loftusa** [comment]
>  I can fwd you some email chains on it if you're interested.

Yeah, that'd be great, wanna understand this a bit better. 

So I guess, given that, I'd personally be a lot less invested in this functionality being there, since you have it with OMNI by default (and you don't have to do procrustes).

**Comment by bdpedigo** [comment]
Forwarded you an email chain.

FWIW there still could be a use case if you don't want to introduce the OMNI bias which will push nodes to be closer together. I do still just think we'd have to think harder about robust ways of doing this alignment in the MASE case

**Comment by loftusa** [comment]
cool, thanks, that's probably true. For me it actually popped up as I was essentially trying to recreate a simpler version of guodong's stuff for the book, so the 'just work with the R matrices' suggestion might be the play. Or just ignore MASE for demonstration purposes and do exclusively OMNI.

Anyway, for the purposes of this issue - we could implement a method in the MASE class where we make the X's and then align them with Procrustes, but not sure if that's more effort than it's worth

**Comment by bdpedigo** [comment]
the nice thing is that you can actually align the decomposed $R$ matrices, which we thought should be easier.

however, for the purposes of this issue, my vote would simply be we close this issue until someone has a compelling case for the right thing to do. currently this feels like something we could reach to implement but it might not be super trustworthy, and given the current code, someone who is familiar with this algorithm could do it themselves with a little bit of work.

**Comment by loftusa** [comment]
Sounds good to me
